File: Phase03/EntryTest.java

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

public class EntryTest {
    private Entry entry;

    @BeforeEach
    public void createObject() throws Exception{
        entry = new Entry("test", 11);
    }

    @Test
    public void getIndexTest(){
        assertEquals(entry.getIndex(),11);
    }
    @Test
    public void toStringTest(){
       
        assertEquals(entry.toString(), "Document Id: test, Index: 11");
    }

    @Test
    public void equalsTest(){
        Entry obj = new Entry("test", 11);
        assertEquals(obj.equals(entry) , true);
        assertEquals(obj.equals("test") , false);
    }

    @Test
    public void hashCodeTest(){
        assertEquals(entry.hashCode() , "test".hashCode());
    }



}

File: app/library.Test/EntryTest.cs
using Xunit;
using System;

namespace library.Test
{
    public class EntryTest
    {
        private Entry entry;

        public EntryTest()
        {
            this.entry = new Entry("test", 11);
        }

        [Fact]
        public void GetIndexTest()
        {
            Assert.Equal(entry.Index, 11);
        }

        [Fact]
        public void ToStringTest()
        {
            Assert.Equal(entry.ToString(), "Document Id: test, Index: 11");
        }

        [Theory]
        [InlineData("test", 11)]
        [InlineData("test", 12)]
        public void EqualsTestTrue(string docName, int index)
        {
            Assert.True(this.entry.Equals(new Entry(docName, index)));
        }

        [Theory]
        [InlineData("wrong test", 11)]
        [InlineData("wrong test", 12)]
        public void EqualsTestFalse(string docName, int index)
        {
            Assert.False(this.entry.Equals(new Entry(docName, index)));
        }

        [Fact]
        public void EqualsTestFalseDifferentType()
        {
            Assert.False(this.entry.Equals(new object()));
            Assert.False(this.entry.Equals("test"));
        }

        [Fact]
        public void GetHashCodeTest()
        {
            Assert.Equal(entry.GetHashCode(), "test".GetHashCode());
        }
    }
}
